src/common/procedure/src/local.rs

```diff
@@ -487,18 +517,35 @@ impl ProcedureManager for LocalManager {
     }
 
     async fn procedure_state(&self, procedure_id: ProcedureId) -> Result<Option<ProcedureState>> {
-        self.manager_ctx.remove_outdated_meta(META_TTL);
-
         Ok(self.manager_ctx.state(procedure_id))
     }
 
     fn procedure_watcher(&self, procedure_id: ProcedureId) -> Option<Watcher> {
-        self.manager_ctx.remove_outdated_meta(META_TTL);
-
         self.manager_ctx.watcher(procedure_id)
     }
 }
 
+struct RemoveOutdatedMetaFunction {
+    manager_ctx: Arc<ManagerContext>,
+    ttl: Duration,
+}
+
+use common_runtime::TaskFunction;
+
+use crate::error::Error;
```

Comment:
nitpick: Could you move them to line 30 so the `cargo fmt` can merge these imports?